paasta_tools/paasta_serviceinit.py

```diff
@@ -143,57 +144,109 @@ def main():
+                    "know how to handle."
+                ).format(
+                    instance_type, compose_job_id(service, instance),
+                ),
+            )
+            return_codes.append(1)
+        else:
+            instance_types_map[instance_type].append(instance)
+
+    remote_run_frameworks = None
+    if len(instance_types_map['adhoc']) > 0:
+        remote_run_frameworks = paasta_remote_run.remote_run_frameworks()
+
+    for instance_type in instance_types:
+        for instance in instance_types_map[instance_type]:
+            try:
+                version = get_deployment_version(
+                    actual_deployments, cluster, instance,
                 )
-            elif instance_type == 'paasta_native':
-                return_code = paasta_native_serviceinit.perform_command(
-                    command=command,
-                    service=service,
-                    instance=instance,
-                    cluster=cluster,
-                    verbose=args.verbose,
-                    soa_dir=args.soa_dir,
+                paasta_print('instance: %s' % PaastaColors.blue(instance))
+                paasta_print('Git sha:    %s (desired)' % version)
+
+                if instance_type == 'marathon':
+                    return_code = marathon_serviceinit.perform_command(
+                        command=command,
+                        service=service,
+                        instance=instance,
+                        cluster=cluster,
+                        verbose=args.verbose,
+                        soa_dir=args.soa_dir,
+                        app_id=args.app_id,
+                        delta=args.delta,
+                        client=clients.marathon(),
+                    )
+                elif instance_type == 'chronos':
+                    return_code = chronos_serviceinit.perform_command(
+                        command=command,
+                        service=service,
+                        instance=instance,
+                        cluster=cluster,
+                        verbose=args.verbose,
+                        soa_dir=args.soa_dir,
+                        client=clients.chronos(),
+                    )
+                elif instance_type == 'paasta_native':
+                    return_code = paasta_native_serviceinit.perform_command(
+                        command=command,
+                        service=service,
+                        instance=instance,
+                        cluster=cluster,
+                        verbose=args.verbose,
+                        soa_dir=args.soa_dir,
+                    )
+                elif instance_type == 'adhoc':
+                    if instance == 'interactive':
+                        raise NotImplementedError
+                    if command != 'status':
+                        raise NotImplementedError
+
+                    paasta_remote_run.remote_run_list(
+                        argparse.Namespace(
```

Comment:
Can you not use the argparser?

Reply:
i'd like to reuse the `remote_run_list` but it takes argparse.Namespace as an argument and i don't want to change it's signature. could you elaborate why i shouldn't use argparse here?

Reply:
It is nice to call our functions consistently using normal keyword arguments. Plus it helps with static validation (vim, pylint, mocking, etc) to make sure we are sending it valid keyword arguments. And finally just general consistency in the codebase.

Reply:
done